Common/gfx/bitmap.cpp

```diff
@@ -221,6 +214,738 @@ void ReadPixelsFromMemory(Bitmap *dst, const uint8_t *src_buffer, const size_t s
+        hdr.h = in->ReadInt32();                   /* bitmap height in pixels (signed integer)  */
+        hdr.colorPlanes = in->ReadInt16();         /* number of color planes (must be 1)  */
+        hdr.bpp = in->ReadInt16();                 /* color depth, number of bits per pixel */
+        hdr.compression = (BMP_Info_Compression) in->ReadInt32();         /* compression method being used. */
+        hdr.mapSizeImage = in->ReadInt32();        /* image size of raw bitmap data; a dummy 0 can be given for BI_RGB bitmaps. */
+        hdr.xPxPerMeter = in->ReadInt32();         /* horizontal resolution. (pixel per metre, signed integer) (0xB12 = 72 dpi) */
+        hdr.yPxPerMeter = in->ReadInt32();         /* vertical resolution. (pixel per metre, signed integer)  */
+
+        hdr.colorsCount = in->ReadInt32();          /* number of colors in the color palette */
+        hdr.importantColorsCount = in->ReadInt32(); /* number of important colors used */
+
+        if(hdr.compression != kBI_BitFields) {
+            read_palette(hdr.offsetBits - 54, pal, in, false);
+        }
+    } else {
+        return nullptr; // unsupported bitmap type
+    }
+
+    bpp = hdr.bpp;
+    w = hdr.w;
+    h = std::abs(hdr.h);
+
+    if(hdr.compression == kBI_BitFields) {
+        int rMask = in->ReadInt32();
+        int gMask = in->ReadInt32();
+        int bMask = in->ReadInt32();
+
+        if ((bMask == 0x001f) && (rMask == 0x7C00)) {
+            bpp = 15;
+        } else if ((bMask == 0x001f) && (rMask == 0xF800)) {
+            bpp = 16;
+        } else if ((bMask == 0x0000FF) && (rMask == 0xFF0000)) {
+            bpp = 32;
+        } else {
+            /* Unrecognised bit masks/depth, refuse to load. */
+            return nullptr;
+        }
+    }
+
+    Bitmap* bmp = BitmapHelper::CreateBitmap(w,h,bpp);
+    if (!bmp) {
+        return nullptr;
+    }
+
+    switch (hdr.compression) {
+        case kBI_RGB_None:
+            read_image(in, bmp, hdr);
+            break;
+        case kBI_RLE8:
+            read_RLE8_compressed_image(in, bmp, hdr);
+            break;
+        case kBI_RLE4:
+            read_RLE4_compressed_image(in, bmp, hdr);
+            break;
+        case kBI_BitFields:
+            read_bitfields_image(in, bmp, hdr);
+            break;
+        default:
+            bmp->Destroy();
+            return nullptr;
+    }
+
+    int dest_depth = color_load_depth(bpp, get_color_depth(),  false);
+
+    if (dest_depth != bpp) {
+        /* restore original palette except if it comes from the bitmap */
+        if ((bpp != 8) && (!want_palette))
+            pal = NULL;
+
+        if (bmp) {
+            Bitmap* tmp_bmp = BitmapHelper::CreateRawBitmapOwner(
+                    _fixup_loaded_bitmap(bmp->GetAllegroBitmap(), pal, dest_depth));
+            Bitmap* tmp_bmp2 = BitmapHelper::CreateBitmapCopy(tmp_bmp);
+
+            bmp->Destroy();

```

Comment:
Calling Destroy here is not enough, as this method only deletes the internal data, but not the `bmp` object itself. You need to do `delete bmp` in order to completely deallocate Bitmap object.

Reply:
good catch! I adjusted this too.

Reply:
thanks! Fixes added, [comparison here](https://github.com/adventuregamestudio/ags/compare/c16711bae808d5145c8561d98e3ea35a835fd0fd..42b9759d880f15bc4103ed9e6d62915c51f70916)